File: matiasmasso/Mat.NET/DT2/DTOProductDimensions.vb
Public Class DTOProductDimensions
    Property Hereda As Boolean
    Property NoDimensions As Boolean
    Property KgNet As Decimal
    Property KgBrut As Decimal
    Property M3 As Decimal
    Property DimensionAlto As Decimal
    Property DimensionAncho As Decimal
    Property DimensionLargo As Decimal
    Property InnerPack As Integer
    Property OuterPack As Integer
    Property ForzarInnerPack As Boolean
    Property CodiMercancia As DTOCodiMercancia
    Property PackageEan As DTOEan


    Shared Function DimensionLess() As DTOProductDimensions
        Dim oRetVal As New DTOProductDimensions()
        With oRetVal
            .NoDimensions = True
            .Hereda = False
        End With
        Return oRetVal
    End Function


End Class

File: matiasmasso/Mat.NET/DTO/DTOProductDimensions.cs
namespace DTO
{
    public class DTOProductDimensions
    {
        public bool Hereda { get; set; }
        public bool NoDimensions { get; set; }
        public decimal KgNet { get; set; }
        public decimal KgBrut { get; set; }
        public decimal M3 { get; set; }
        public decimal DimensionAlto { get; set; }
        public decimal DimensionAncho { get; set; }
        public decimal DimensionLargo { get; set; }
        public int InnerPack { get; set; }
        public int OuterPack { get; set; }
        public bool ForzarInnerPack { get; set; }
        public DTOCodiMercancia CodiMercancia { get; set; }
        public DTOEan PackageEan { get; set; }


        public static DTOProductDimensions DimensionLess()
        {
            DTOProductDimensions oRetVal = new DTOProductDimensions();
            {
                var withBlock = oRetVal;
                withBlock.NoDimensions = true;
                withBlock.Hereda = false;
            }
            return oRetVal;
        }
    }
}
